src/main/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemMigrateSearchCriteriaTask.java

```diff
@@ -0,0 +1,84 @@
+import uk.gov.hmcts.sptribs.systemupdate.service.CcdManagementException;
+import uk.gov.hmcts.sptribs.systemupdate.service.CcdSearchCaseException;
+import uk.gov.hmcts.sptribs.systemupdate.service.CcdSearchService;
+import uk.gov.hmcts.sptribs.systemupdate.service.CcdUpdateService;
+
+import java.util.List;
+
+import static org.elasticsearch.index.query.QueryBuilders.boolQuery;
+import static org.elasticsearch.index.query.QueryBuilders.existsQuery;
+import static uk.gov.hmcts.sptribs.systemupdate.event.SystemMigrateSearchCriteria.SYSTEM_MIGRATE_SEARCH_CRITERIA;
+
+@Component
+@Slf4j
+public class SystemMigrateSearchCriteriaTask implements Runnable {
+
+    private final AuthTokenGenerator authTokenGenerator;
+
+    private final CcdSearchService ccdSearchService;
+
+    private final CcdUpdateService ccdUpdateService;
+
+    private final IdamService idamService;
+
+    @Autowired
+    public SystemMigrateSearchCriteriaTask(AuthTokenGenerator authTokenGenerator, CcdSearchService ccdSearchService,
+                                           CcdUpdateService ccdUpdateService, IdamService idamService) {
+        this.authTokenGenerator = authTokenGenerator;
+        this.ccdSearchService = ccdSearchService;
+        this.ccdUpdateService = ccdUpdateService;
+        this.idamService = idamService;
+    }
+
+    @Override
+    public void run() {
+        final User user = idamService.retrieveSystemUpdateUserDetails();
+        final String serviceAuth = authTokenGenerator.generate();
+
+        try {
+            final BoolQueryBuilder query =
+                boolQuery()
+                    .must(boolQuery()
+                        .mustNot(existsQuery("data.SearchCriteria"))
+                    );
+
+            final List<CaseDetails> casesToMigrateSearchCriteria =
+                ccdSearchService.searchForAllCasesWithQuery(query, user, serviceAuth);
+            log.info("Cases:" + casesToMigrateSearchCriteria.size());
+
+            for (final CaseDetails caseDetails : casesToMigrateSearchCriteria) {
+                triggerSystemMigrateSearchCriteriaTask(user, serviceAuth, caseDetails);
+            }
+
+            log.info("System migrate search criteria scheduled task complete.");
+        } catch (final CcdSearchCaseException e) {
+            log.error("System migrate search criteria schedule task stopped after search error", e);
+        } catch (final CcdConflictException e) {
+            log.info("System migrate search criteria schedule task stopping "
+                + "due to conflict with another running task"
+            );
+        }
+    }
+
+    private void triggerSystemMigrateSearchCriteriaTask(User user, String serviceAuth, CaseDetails caseDetails) {
+        try {
+            log.info("Submitting System Migrate Search Criteria Event for Case {}", caseDetails.getId());
+            ccdUpdateService.submitEvent(caseDetails.getId(), SYSTEM_MIGRATE_SEARCH_CRITERIA, user, serviceAuth);
```

Comment:
We could migrate the case location (region, base location), case management category and supplementary data as well as part of this migration, so that we could avoid the overhead of running separate queries for them.

Reply:
Sure - values as per the ones you gave the for original SQL query @rajaram-nagarajan ?

Reply:
Have actioned